explore/MultiTaskClassification.py

```diff
@@ -0,0 +1,461 @@
+
+# coding: utf-8
+
+# # Multi-task classification: Random forests and (Multitask)Lasso
+
+# In[1]:
+
+import os
+import urllib
+import random
+import warnings
+import itertools
+
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+from sklearn import preprocessing
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import accuracy_score
+from sklearn.metrics import precision_score
+from sklearn.metrics import confusion_matrix
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import StandardScaler, binarize
+from sklearn.feature_selection import SelectKBest
+from statsmodels.robust.scale import mad
+
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.linear_model import Lasso
+from sklearn.linear_model import MultiTaskLasso
+
+
+# In[2]:
+
+get_ipython().magic('matplotlib inline')
+plt.style.use('seaborn-notebook')
+
+
+# ## Load Data
+
+# In[3]:
+
+get_ipython().run_cell_magic('time', '', "path = os.path.join('..','download', 'expression-matrix.tsv.bz2')\nX = pd.read_table(path, index_col=0)")
+
+
+# In[4]:
+
+get_ipython().run_cell_magic('time', '', "path = os.path.join('..','download', 'mutation-matrix.tsv.bz2')\nY = pd.read_table(path, index_col=0)")
+
+
+# In[5]:
+
+#functions for data preprocessing and figure labels
+
+def fs_mad(x, y):
+    """
+    Get the median absolute deviation (MAD) for each column of x
+    """
+    scores = mad(x) 
+    return scores, np.array([np.NaN]*len(scores))
+# select the top features with the highest MAD
+feature_select = SelectKBest(fs_mad, k=2000)
+
+pipeline = make_pipeline(
+    feature_select,  # Feature selection
+    StandardScaler(),  # Feature scaling
+    )
+
+def autolabel(rects,ay):
+    # attach some text labels
+    for rect in rects:
+        width = rect.get_width()
+        ay.text(1.05*width, rect.get_y() + rect.get_height()/2., 
+                '%.2f' % width,
+                ha='center', va='bottom')
+        
+
+
+# In[6]:
+
+# Mono-task classifiers, random forests and Lasso. Lasso results will be binarized. 
+
+weights = {0:.4,1:.6}
+
+Classifiers_mono = {
+    "Random forest": RandomForestClassifier(max_depth=100, min_samples_split=2, n_estimators=10, random_state=0,class_weight = weights),
+    "Lasso": Lasso(alpha=0.1),
+}
+
+
+# In[7]:
+
+# Agents and targeted genes

```

Comment:
These dictionaries are really useful. Thanks for doing this reconnaissance work. You may want to consider making a TSV file for this data, so you don't have to code it in the notebook. I could imagine a table with three columns: entrez_gene_id, gene_symbol, and drug.